lib/biz/calculation/active.rb

```diff
@@ -10,16 +10,24 @@ def initialize(schedule, time)
       end
 
       def result
-        schedule.intervals.any?      { |interval| interval.contains?(time) } \
-          && schedule.breaks.none?   { |brake| brake.contains?(time) } \
-          && schedule.holidays.none? { |holiday| holiday.contains?(time) }
+        in_hours? && active?
       end
 
       private
 
       attr_reader :schedule,
                   :time
 
+      def in_hours?
+        schedule.intervals.any? { |interval| interval.contains?(time) } ||
+          schedule.shifts.any? { |shift| shift.contains?(time) }
```

Comment:
I just realized there's an inconsistency between the behavior of this check and how shifts work in the other calculations. Given that intervals are disregarded for any days where shifts are configured, this logic should reflect that behavior.

If you're wondering why this calculation doesn't use period generation like the other calculations, the reason is that period generation is relatively slow. So, instead of using it for this relatively straightforward calculation, I wrote this optimized version. Unfortunately, here's a case where a separate implementation comes back to bite us.

I'll revise and post an update when I'm finished.